### PR TITLE
fix(#746): route the single-site split-CTM path through the 2x2 recipe

### LINKED ISSUE #746: Route the single-site split-CTM path through the 2x2 plaquette projector (the real #726 fix)
Follow-up to #726, which is currently only *guarded*, not fixed.

## What #726 established

The `1x1` corner-pair (Fishman) projector collapses the CTM environment to rank-1 corners. It is **not** a split-CTM defect — the fused and split single-site paths call the same `_compute_projector_tensor`, and both collapse identically:

| recipe | C1 spectrum (χ=8) | rank |
|---|---|---|
| split 2×2 | `1, 0.128, 0.127, 0.016, 0.0021, 0.0020` | 6 |
| fused 2×2 | identical, digit-for-digit | 6 |
| split 1×1 | `1, 0, 0, 0, 0, 0` | 1 |
| fused 1×1 | `1, 0, 0, 0, 0, 0` | 1 |

**Mechanism:** `_ctm_projector.py:981` forms `M = C1g† C4g`, which is χ×χ — the χ·D² seam is summed away, so `M` sees only the outer χ legs. With `M = U S V^H`, `C1_new = S^½U^H`, so the corner's entire spectrum is `√spec(M)` and `rank(C_new) ≤ rank(C1g)`, which is 1 at cold init. Verified numerically to 7 digits.

**Consequence:** the energy is χ-independent — bit-identical from χ=2 to χ=32 on a physical D=2 SU state (`−0.485880315530`, delta exactly `0.0`). That is a χ_eff=1 mean-field boundary, not a corner transfer matrix. The correct value via 2×2 is `−0.4886385`.

## Why the obvious fixes don't work

- **Seed χ_eff = D²** (the #723 suggestion): refuted directly. `eye(min(χ,D²))`, `eye(χ)`, and random corner seeds all still collapse. The corner seed is not the binding constraint.
- **Retain the fused index in the cross product**: structurally impossible. `C1g` is `(χD², χ)`, so `rank(C1g) ≤ χ` and `= 1` cold, whatever index is left free. The two-corner projector cannot see the D² seam.

The binding constraint is the rank-1 **edge** seed (`_ctm_tensor_init.py:248`, `_split_ctm_tensor_init.py:81,116`) feeding a projector that cannot grow rank. Warm-starting the 1×1 sweep from a *converged* 2×2 fixed point destroys it — energies go to `−1.007` and `+0.508`, both impossible for this state — so the sweep is not merely failing to build a good environment, it actively degrades a correct one.

## Proposed fix

Route the single-site path through the existing, working 2×2 plaquette projector: add a `recipe` parameter to `ctm_split_tensor` (`_split_ctm_tensor_convergence.py:111`) and `ctm_split_tensor_converge_explicit` (`ad_utils.py:1325`), delegating to `_split_ctm_sweep_multisite(..., recipe=recipe)` with a 1-site neighbour map. Verified to work end-to-end under `jax.grad`.

Also in scope: the split edge init should reproduce `δ_{ket=bra}` to match the fused fix (`_ctm_tensor_init.py:236-243` documents the all-ones boundary as "the wrong boundary … traps CTM at a degenerate fixed point"; the fused init was fixed, the split init was not).

## Tests shaped around the symptom (must be revisited)

- `tests/test_regularized_svd.py:244-250` asserts "the boundary corner of a uniform 1-site iPEPS is genuinely rank-1" and routes around `sum(C1**2)` because its gradient is exactly 0. Measured: current 1×1 gives `sum(C1**2) = 1.0` and `|grad| = 0.0` exactly; the 2×2 candidate gives `1.428` and `1.980`.
- `tests/test_split_ctm_doublelayer_projector.py:28` `test_split_corner_init_is_rank1` pins the rank-1 seed as a *requirement*.
- `tests/test_split_ctm_doublelayer_projector.py:126` asserts `chi_I=chi` is "already lossless" — true only because the environment is χ_eff=1.
- `test_split_matches_fused_lossless_chi_I` / `test_fused_to_split_roundtrip` validate split-1×1 against fused-1×1 — a **circular oracle**, since both share the broken projector.

## Not established

- Whether the 1×1 corner-pair projector is repairable in principle, or only replaceable by the 2×2 plaquette. Only shown: the current formulation cannot escape rank collapse.
- The origin of the split-1×1 warm-start non-convergence, as distinct from the rank collapse — possibly a separate defect.
- Whether `SymmetricTensor` inputs behave the same; all measurements are dense.

Related: #723 (same mechanism, fused path), #463 (split-CTM), #726.

> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

### PR DESCRIPTION
**Pull request by yingjerkao**
Closes #746. The **split** half of the rank-1 environment collapse; #765 did the fused half.

## The defect

`ctm_split_tensor` had no recipe knob and always ran the legacy `1x1` single-site moves. The corner-pair projector is shared *verbatim* with the fused path, so this was never a split-CTM defect: `M = C1g^H C4g` is `chi x chi`, the `chi*D²` seam is summed away, so `rank(P) <= rank(C1g)` — which is 1 at the cold rank-1 seed, making rank-1 an absorbing state.

Measured on the D=2 sublattice-rotated Heisenberg simple-update state of #726/#747:

| path | chi=4 | chi=16 | rank(C1) |
|---|---|---|---|
| split `1x1` | 0.49620072949960814 | *bit-identical* | **1** |
| **split `2x2`** | 0.4991254745638001 | 0.49912538701724773 | 4 → 6 |
| fused `2x2` (oracle) | 0.4991253869195439 | 0.4991253869195441 | 4 → 6 |

The split `2x2` corner spectrum reproduces the fused one digit-for-digit (`1, 0.12764, 0.12659, 0.01638, 0.00208, 0.00202`) and the energies agree to 1e-10 at chi=16.

## The fix

Three commits.

**1 — forward.** `recipe: str = "2x2"` on `ctm_split_tensor` and `ctm_split_tensor_converge_explicit`, delegating to `_split_ctm_sweep_multisite` with `SINGLE_SITE_NEIGHBORS`: a uniform 1-site lattice *is* the multisite path with a self-referential neighbour map, so the working 2x2 plaquette projector applies verbatim. Convergence loop, `min_iter` floor and return contracts untouched. `"1x1"` kept for bisection.

**2 — AD.** The forward alone left the fix unreachable from production: `make_ctm_energy_fn` and `validate_split_ctm_config` *rejected* every `gs_recipe` but `"1x1"` on the single-site split branch (#726 made it raise rather than lie). Now `recipe` is threaded through `_split_step` (the fixed-point map whose Jacobian drives the implicit backward), the `custom_vjp` static tuple, `converge_split_env`, `ctm_energy_split_{explicit,implicit}` and the optimizer's `_split_forward`. Both guards drop.

Gradient validation (D=2 SU state, chi=4, chi_I=4):

- implicit (Neumann) vs explicit (unrolled) backward: **7.1e-10** relative
- directional FD along `g/|g|` vs analytic `g·v`: **5.3e-5**, stable across eps=1e-4..1e-6 — so the residual is CTM fixed-point accuracy, not a gradient error
- warm-start seed-independence: **|dE| = 0.0** exactly, gradients to 1.9e-13

**3 — kernel-parity fixture.** See below.

## Test changes

Every test that moved was shaped around the collapse, as #746 predicted.

- **`test_split_matches_fused_lossless_chi_I`** — the circular oracle. The temporary `recipe="1x1"` pin that bridged #723 and #746 is dropped; both sides now run 2x2 and agree to **2.5e-16 / 9.0e-16 / 5.0e-16**. It is a real oracle now instead of two collapsed paths agreeing.
- **`test_split_production_chi_I_converges_to_lossless`** asserted the interlayer error is "~0, not merely shrinking" (`< 1e-6`). That was the collapse: at chi_eff=1 there is nothing to truncate, and 1x1 returns a bit-identical energy at chi_I=4 and chi_I=8. On 2x2 the error is real and shrinks (7.24e-4 → 5.64e-4 for chi 4 → 6).
- **`test_split_min_iter_floor_blocks_early_break`** counted calls to `_split_ctm_tensor_sweep`, which the default path no longer calls — it would have counted zero and passed for the wrong reason.
- **`test_symmetric_energy_matches_dense`** ran chi_I=4 against chi=8; now runs the production chi_I=chi. See #766.
- **`test_split_ctm_chi_frozen_726`** guards invert: the default config must now *run* and return a rank>1 corner, including the `gs_num_steps=0` evaluate-only route that used to bypass a loss-only guard (PR #749 review). `gs_recipe="1x1"` keeps its own test and must still collapse.
- **Two `test_split_ctm_fuse_flag` tests pinned to `1x1`** — see #767.

`test_ctm_723_single_site_collapse.py` and the new `test_split_ctm_746_single_site_collapse.py` were both absent from `_FILE_MARKERS`, so `-m core` deselected them and neither ran in a required job. Added as `core` (~50s each, D=2, chi<=16 dense).

## Also fixes 16 failures already on `main`

`test_split_ctm_2site.py` is `algorithm`-marked, so `-m core` deselects it and #765's required checks never ran it. It has been failing on `main` at 988c2a8 since #765 merged.

`_split_env_and_fused_env` claimed to return a "matched pair" but converged the two envs *independently*. Callers assert a tensor-level identity between a split kernel and its fused counterpart, which needs the same environment on both sides — and that only held because both entry points ran `1x1` and landed on the same rank-1 corner. #723 moved the fused side to 2x2 and left split on 1x1, mismatching the pair.

Moving split to 2x2 does not fix it either: the split env carries a `chi_I` truncation the fused env does not, and on these random site tensors the 2x2 forward does not converge element-wise (#767). On seed=5, chi=6 the fused enlarged corner keeps a degenerate pair `0.1654, 0.1654` where split splits it to `0.1605, 0.1748`, and a structural zero (4.3e-19) becomes 8.8e-3. Both are legitimate environments, just not the same one.

The split env is now built as the *exact factorization* of the fused env (`fused_env_to_split` at the lossless `chi_I = chi*D`). The premise holds by construction, on the production 2x2 recipe rather than the collapsed one. All 27 pass.

## Filed while working

- **#766** — split 2x2 gives different dense vs symmetric environments when `chi_I << chi` on random tensors. Pre-existing: the 2-site path reproduces the gap to the digit; 1x1 had made `chi_I` inert. Physical states are clean at every `chi_I` from 2 to 16.
- **#767** — the gauge-fixed AD forward silently returns a non-converged environment. On random tensors the 2x2 sweep does not converge element-wise (per-sweep diff 0.5..0.85 after 80 sweeps against conv_tol=1e-12); the already-shipped 2-site path shares this. Physical states converge to 1.3e-14.

## Verification

Run locally on CPU: the six split-CTM suites, `test_ad_utils`, `test_regularized_svd`, `test_ctm_recipe_2x2_consistency`, and the two collapse-guard files under `-m core`. All green.

> 🤖 **AI-generated PR** — written by Claude Code, posted by @yingjerkao.

### CODE REVIEW

**Comment by yingjerkao** [comment]
> 🤖 **AI-generated comment** — written by Claude Code, posted by @yingjerkao.

**Deliberately out of scope: the split edge init (#769).**

#746 also listed "the split edge init should reproduce `δ_{ket=bra}` to match the fused fix". This PR does not do it, on purpose.

It is a real defect — measured, D=2 chi=4, contracting the split ket/bra edge pair over its interlayer bond against the fused `T1`:

```
split -> merged T1[0,:,:,0]:   fused init T1[0,:,:,0]:
[[1. 1.]                       [[1. 0.]
 [1. 1.]]                       [0. 1.]]
max|merged - fused| = 1.0
```

So the split path builds `1_ket ⊗ 1_bra` where the fused path builds `δ_{ket=bra}` — the boundary `_ctm_tensor_init.py` documents as one that "traps CTM at a degenerate fixed point".

I prototyped the corrected init and measured it against the #767 non-convergence. It fixes one fixture (seed=11 chi=4: last-sweep diff 6.5e-1 → 4.1e-2) and does nothing for the other two, so it is **not** a convergence fix and the two defects are independent.

Folding it in would perturb every split environment and require its own regression pass over the split suites, for no demonstrated benefit here. Tracked as #769 with the measurements and the representable `chi_init=1` form of the fix.

File: src/tenax/algorithms/_split_ctm_tensor_convergence.py [hunk 1]
```diff
@@ -117,6 +118,7 @@ def ctm_split_tensor(
     renormalize: bool = True,
     min_iter: int = 2,
     return_info: bool = False,
+    recipe: str = "2x2",
```
[the remark this answers is not in the document]
Comment (yingjerkao) [reply src/tenax/algorithms/_split_ctm_tensor_convergence.py:121]: > 🤖 **AI-generated comment** — written by Claude Code, posted by @yingjerkao.

Re-verified against `main`, one partly fixed and one untouched — both now tracked as **#795**.

**The README comment landed on line 26** (which correctly describes the `"2x2"` default and marks `"1x1"` bisection-only) but **missed line 23**, where the QR-CTMRG projector is still presented as usable under AD "via `gs_recipe="1x1"` + `gs_projector_method="qr"`". So the collapsing recipe is still recommended, just from a different bullet. The issue also asks the question the edit alone would paper over: whether `gs_projector_method="qr"` works under `"2x2"` at all — if it does not, that is a feature-level gap rather than a docs bug.

**The recipe fallback is unchanged**: `_split_ctm_energy_ad.py` still takes an unguarded `else` into `_split_ctm_tensor_sweep`, so `recipe="2X2"` silently runs the rank-collapsing sweep while `ctm_split_tensor` and the explicit-AD entry point raise on the same input.

Resolving here in favour of #795 since this PR is merged.
